apps/projects/app/components/Panel/NewProject/NewProject.js

```diff
@@ -1,41 +1,87 @@
 import React, { useEffect, useRef, useState } from 'react'
 import PropTypes from 'prop-types'
 import styled from 'styled-components'
-import { Button, GU, IconSearch, Info, RadioList, Text, TextInput, theme } from '@aragon/ui'
+import { Button, GU, IconSearch, Info, RadioList, Text, TextInput, textStyle, useTheme } from '@aragon/ui'
 import { GET_REPOSITORIES } from '../../../utils/gql-queries.js'
-import { LoadingAnimation } from '../../Shared'
+import { IconGitHub, LoadingAnimation } from '../../Shared'
 import { Query } from 'react-apollo'
 import { useAragonApi } from '../../../api-react'
 import { usePanelManagement } from '../../Panel'
 import { toHex } from 'web3-utils'
 import noResultsSvg from '../../../assets/noResults.svg'
+import { FormField, FieldTitle } from '../../Form'
+import { STATUS } from '../../../utils/github'
 
+const RepoList = ({
+  filter,
+  handleClearSearch,
+  onRepoSelected,
+  repoArray,
+  repoSelected,
+  visibleRepos
+}) => {
+  const theme = useTheme()
 
-const NewProject = () => {
-  const { api, appState: { repos } } = useAragonApi()
-  const { closePanel } = usePanelManagement()
+  if (visibleRepos.length) return (
+    <StyledRadioList
+      items={repoArray}
+      selected={repoSelected}
+      onChange={onRepoSelected(repoArray)}
+    />
+  )
+
+  if (filter) return (
+    <RepoInfo>
+      <img css={`margin-bottom: ${2 * GU}px`} src={noResultsSvg} alt=""  />
+      <Text.Block style={{ fontSize: '28px', marginBottom: '8px' }}>
+        No results found.
+      </Text.Block>
+      <Text.Block>
+        We can&#39;t find any items mathing your search.
+      </Text.Block>
+      <Button
+        size="mini"
+        onClick={handleClearSearch}
+        css={`
+          margin-left: 8px;
+          border: 0;
+          box-shadow: unset;
+          padding: 4px;
+        `}
+      >
+        <Text size="small" color={`${theme.link}`}>
+          Clear Filters
+        </Text>
+      </Button>
+    </RepoInfo>
+  )
+
+  return (
+    <RepoInfo>
+      <Text>No more repositories to add...</Text>
+    </RepoInfo>
+  )
+}
+RepoList.propTypes = {
+  filter: PropTypes.string.isRequired,
+  handleClearSearch: PropTypes.func.isRequired,
+  onRepoSelected: PropTypes.func.isRequired,
+  repoArray: PropTypes.array.isRequired,
+  repoSelected: PropTypes.number.isRequired,
+  visibleRepos: PropTypes.array.isRequired,
+}
+
+const RepoQuery = ({ onRepoSelected, repoSelected, setRepoSelected }) => {
+  const theme = useTheme()
   const [ filter, setFilter ] = useState('')
-  const [ project, setProject ] = useState()
-  const [ repoSelected, setRepoSelected ] = useState(-1)
+  const searchRef = useRef(null)
```

Comment:
Small note: I don't think you need the `null` here. `useRef()` will initialize just fine without it

Reply:
I think so - fixed.